services/workflows-service/src/transaction/transaction.mapper.ts

```diff
@@ -59,110 +45,119 @@ export class TransactionEntityMapper {
-      statusReason: record.transactionStatusReason || undefined,
-      baseAmount: record.transactionBaseAmount,
-      baseCurrency: record.transactionBaseCurrency,
-      projectId: record.projectId,
-      direction: record.transactionDirection || undefined,
-      reference: record.transactionReference || undefined,
-      originator: record.counterpartyOriginatorId
-        ? {
-            id: record.counterpartyOriginatorId,
-            correlationId: '', // TODO: Add the required correlationId property here (should come from the countryparty table)
-          }
-        : undefined,
 
-      beneficiary: record.counterpartyBeneficiaryId
-        ? {
-            id: record.counterpartyBeneficiaryId,
-            correlationId: '', // TODO: Add the required correlationId property here
-          }
-        : undefined,
+      project: { connect: { id: projectId } },
+    } as const satisfies Omit<
+      Prisma.TransactionRecordCreateArgs['data'],
+      | 'createdAt'
+      | 'updatedAt'
+      | 'id'
+      | 'originatorIpAddress'
+      | 'originatorGeoLocation'
+      | 'originatorUserAgent'
+      | 'auditTrail'
+      | 'riskScore'
+      | 'endUserId'
+      | 'businessId'
+    >;
+  }
 
-      payment: {
-        method: record.paymentMethod || undefined,
-        type: record.paymentType || undefined,
-        channel: record.paymentChannel || undefined,
-        issuer: record.paymentIssuer || undefined,
-        gateway: record.paymentGateway || undefined,
-        acquirer: record.paymentAcquirer || undefined,
-        processor: record.paymentProcessor || undefined,
-      },
-      product: {
-        name: record.productName || undefined,
-        description: record.productDescription || undefined,
-        price: record.productPrice || undefined,
-        sku: record.productSku || undefined,
-        id: record.productId || undefined,
-      },
-      // business: {
-      //   id: record.businessId || undefined,
-      // },
-      tags: record.tags,
-      reviewStatus: record.reviewStatus || undefined,
-      reviewerComments: record.reviewerComments || undefined,
-      riskScore: record.riskScore || undefined,
-      regulatoryAuthority: record.regulatoryAuthority || undefined,
-      additionalInfo: record.additionalInfo,
-      createdAt: record.createdAt,
-      updatedAt: record.updatedAt,
-      auditTrail: record.auditTrail,
-      unusualActivityFlags: record.unusualActivityFlags,
-    };
+  private static getCounterpartyCreateInput({
+    counterpartyInfo,
+    projectId,
+  }: {
+    counterpartyInfo: CounterpartyInfo | undefined;
+    projectId: string;
+  }):
+    | Prisma.TransactionRecordCreateInput['counterpartyOriginator']
+    | Prisma.TransactionRecordCreateInput['counterpartyBeneficiary'] {
+    if (counterpartyInfo?.businessData && counterpartyInfo?.endUserData) {
+      throw new Error('Counterparty must have either business or end user data, not both.');
```

Comment:
its should throw 400 validation error, isnt?

Reply:
Correct, I'll fix it